test/io.jl

```diff
@@ -41,13 +41,18 @@ module TestIO
                    B = 'a':'c',
                    C = ["A", "B", "C"],
                    D = CategoricalArray('a':'c'),
-                   E = NullableCategoricalArray(["A", "B", "C"]),
-                   E = Vector{?Int}(1:3),
-                   F = nulls(3),
-                   G = fill(null, 3))
+                   E = CategoricalArray(["A", "B", null]),
+                   F = Vector{?Int}(1:3),
+                   G = nulls(3),
+                   H = fill(null, 3))
 
-    answer = Sys.WORD_SIZE == 64 ? 0x3bd4a4a099ae8aac : 0x5ecdc4bb
-    @test hash(sprint(printtable, dt)) == answer
+    DRT = CategoricalArrays.DefaultRefType
+    @test sprint(printtable, dt) == """
+                                    "A","B","C","D","E","F","G","H"
+                                    1,"a","A","CategoricalArrays.CategoricalValue{Char,$DRT} 'a'","CategoricalArrays.CategoricalValue{String,$DRT} "A"","1",NULL,NULL
```

Comment:
How would you like the CategoricalValue printing to be formatted? In both master and the nl/null branch of the Nullable -> Null PR in CategoricalArrays I get the following printing behavior in 0.7

```
julia> x = CategoricalVector(["s1", "s2", "s3"])
3-element CategoricalArrays.CategoricalArray{String,1,UInt32}:
 CategoricalArrays.CategoricalValue{String,UInt32} "s1"
 CategoricalArrays.CategoricalValue{String,UInt32} "s2"
 CategoricalArrays.CategoricalValue{String,UInt32} "s3"

julia> show(x)
CategoricalArrays.CategoricalValue{String,UInt32}["s1", "s2", "s3"]
julia> showcompact(x)
CategoricalArrays.CategoricalValue{String,UInt32}["s1", "s2", "s3"]
julia> print(x)
CategoricalArrays.CategoricalValue{String,UInt32}["s1", "s2", "s3"]
```